src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -419,9 +419,6 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
         _PreNavigateHelper();
 
-        // Add an event handler for when the user wants to delete a profile.
-        profile.DeleteProfile({ this, &MainPage::_DeleteProfile });
-

```

Comment:
Ok wait. So, setting `DeleteProfile` multiple times would _append_  an event handler? That seems really unintuitive... wtf. Good find!

Reply:
That's actually just how winrt events work. Every call appends another handler to the list of handlers, until revoked. It's basically the same as 

```csharp
profile.DeleteProfile += new EventHandler(this._DeleteProfile);
```